heron/packing/src/java/com/twitter/heron/packing/binpacking/FirstFitDecreasingPacking.java

```diff
@@ -332,7 +338,7 @@ private int allocateNewContainer(ArrayList<Container> containers) {
    * @param instanceRAM The RAM allocated to the instance
    * @return true if the instance is valid, false otherwise
    */
-  protected boolean isValidInstance(long instanceRAM) {
+  protected boolean isValidInstance(long instanceRAM, double instanceCpu, long instanceDisk) {
```

Comment:
PackingPlan has an inner class Resource, which is a tuple of these 3 objects. Should we make that an outer class and leverage it here?